### PR TITLE
camera functions and zone collision data

### PR DESCRIPTION
**Pull request by Unnunu**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀 This PR matches 7 functions (+0.21%) on `us`.

✅ This PR fixes 5 warnings!

File: src/camera.c [hunk 1]
```diff
@@ -1,24 +1,291 @@
...
-void create_camera_leadplayer_matrix(Camera* camera);
+CameraControlSettings* test_ray_zone(f32 posX, f32 posY, f32 posZ, Collider** zone) {
+    f32 hitX, hitY, hitZ, hitDepth, nX, nY, nZ;
+    s32 zoneID;
 
-INCLUDE_ASM(s32, "camera", test_ray_zone);
+    hitDepth = 32767.0f;
+    zoneID = test_ray_zones(posX, posY, posZ, 0, -1.0f, 0, &hitX, &hitY, &hitZ, &hitDepth, &nX, &nY, &nZ);
+    if (zoneID >= 0) {
+        if (zone) {
```
Comment (ethteck) [inline src/camera.c:13]: if zone != NULL

File: src/camera.c [hunk 2]
```diff
@@ -1,24 +1,291 @@
...
 
+s32 func_800325E4(f32, f32, f32, f32, f32, f32, f32, f32, f32*, f32*, f32*);
 INCLUDE_ASM(s32, "camera", func_800325E4);
 
-INCLUDE_ASM(s32, "camera", func_800328A4, Camera* camera, f32 arg1, f32 arg2);
+s32 func_800328A4(CameraControlSettings* camSettings, f32 x, f32 z) {
+    f32 product1, product2;
+    f32 delta, p1, p2, p3, p4;
+
+    if (!camSettings) {
```
Comment (ethteck) [inline src/camera.c:32]: camSettings != NULL (since it's not a bool)

File: src/camera.c [hunk 3]
```diff
@@ -1,24 +1,291 @@
...
+    }
+    return 0;
+}
+
+void func_80032970(Camera* camera, f32 arg1) {
+    f32 stickX;
+    f32 deltaLeadAmount;
+    s32 flags = camera->flags & CAM_FLAG_1000;
+    s32 a2 = flags != 0;
+
```
Comment (ethteck) [inline src/camera.c:68]: currentController != NULL

File: src/camera.c [hunk 4]
```diff
@@ -1,24 +1,291 @@
...
+    f32 f24, f22, cosYaw, sinYaw;
+
+    rotationRad = camera->trueRotation.x / 180 * PI;
+    leadAmount = camera->leadAmount;
+    s2 = 0;
+    cos_rad(rotationRad);
+    sin_rad(rotationRad);
+    settings3 = settings = test_ray_zone(camera->targetPos.x, camera->targetPos.y + 10.0f, camera->targetPos.z, NULL);
+
+    if (settings) {
```
Comment (ethteck) [inline src/camera.c:185]: != NULL (same applies to the rest of the code, I'll stop leaving this comment)

File: src/evt/cam_api.c [hunk 5]
```diff
@@ -545,9 +545,88 @@ ApiStatus WaitForCam(Evt* script, s32 isInitialCall) {
...
+        camera->controlSettings.boomPitch = boomPitch;
+        camera->controlSettings.viewPitch = viewPitch;
+        camera->moveSpeed = moveSpeed;
+        camera->unk_506 = 1;
+        camera->followPlayer = TRUE;
+        camera->panPhase = 0;
+        return ApiStatus_BLOCK;
+    }
+
+    if (camera->interpAlpha >= 1) {
```
Comment (ethteck) [inline src/evt/cam_api.c:581]: `> 0` might make more sense here (not sure if it'll optimize the same though)

File: src/evt/cam_api.c [hunk 6]
```diff
@@ -545,9 +545,88 @@ ApiStatus WaitForCam(Evt* script, s32 isInitialCall) {
...
```
Comment (ethteck) [inline src/evt/cam_api.c:624]: same

**Review by ethteck — CHANGES_REQUESTED** [review]
great shtuff

File: src/evt/cam_api.c (hunk 5 above)
Comment (Unnunu) [reply src/evt/cam_api.c:581]: interpAlpha is float. Should I put 1.0f instead of 1?

Comment (ethteck) [reply src/evt/cam_api.c:581]: ohhh then that will not work, right. yes 1.0f please :)

**Review by nanaian — APPROVED** [review]

**Pull request merged**